### PR TITLE
As/tv os

### PR DESCRIPTION
**Pull request by asowers1**
# Summation of pull request

This PR is intends to add tvOS support to ConsistencyManager-iOS (which may have a confusing naming scheme if made multi-platform?)

# Project additions

ConsistencyManager project:
- Adds appletvos and appletvsimulator as supported platforms
- Adds TARGETED_DEVICE_FAMILY 3 (for tvOS) in project file

SampleApp / ConsistencyManagerDemo project:
- Adds ConsistencyManagerDemo-tvOS target to ConsistencyManagerDemo project
- Ports ConsistencyManagerDemo to ConsistencyManagerDemo-tvOS
- Adds ConsistencyManagerDemoTests files to ConsistencyManagerDemo-tvOSTests target membership

Cocoapods:
- Suggests 5.1.0 pod release in podspec
- Adds tvos requirements to podspec

Misc:
- Some xibs opened with Xcode 8.2.1; automatically modified xib files

# Demo app for tvOS:

simulator screen shot jan 7 2017 5 21 36 pm
simulator screen shot jan 7 2017 5 21 35 pm
simulator screen shot jan 7 2017 5 21 31 pm

# Cool tidbits
Per http://promisekit.org/news/2016/08/Multiplatform-Single-Scheme-Xcode-Projects/, including tvOS support in the ConsistencyManager target.

### CODE REVIEW

**Comment by plivesey** [comment]
So first of all, thanks for the PR. Good point with `-iOS`. Created https://github.com/linkedin/ConsistencyManager-iOS/issues/50.

**Comment by plivesey** [comment]
Going to pull this down locally and take a look at the changes.

**Comment by asowers1** [comment]
👌

**Comment by plivesey** [comment]
Going to look over the code. Mostly looks good but a couple of comments so far:
1) In the demo app, there's still a scheme for `ConsistencyManager` which doesn't do anything anymore (UI shows it as a cog). Can we delete that?
2) In the demo app, there's some code which is exactly the same. Can we move that to a shared folder or something and just add it to both targets? I think just the models and the `Network` folder.

File: README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 [![Build Status](https://travis-ci.org/linkedin/ConsistencyManager-iOS.svg?branch=master)](https://travis-ci.org/linkedin/ConsistencyManager-iOS)
 [![codecov](https://codecov.io/gh/linkedin/ConsistencyManager-iOS/branch/master/graph/badge.svg)](https://codecov.io/gh/linkedin/ConsistencyManager-iOS)
 [![GitHub release](https://img.shields.io/github/release/linkedin/ConsistencyManager-iOS.svg?maxAge=86400)](https://github.com/linkedin/ConsistencyManager-iOS/releases)
+![Platforms](https://img.shields.io/badge/platforms-iOS%20%7C%20tvOS%20%7C%20macOS-blue.svg)
```
Comment (plivesey) [inline README.md:6]: Nice. Thanks.

File: SampleApp/ConsistencyManagerDemo-tvOSUITests/ConsistencyManagerDemo_tvOSUITests.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//  ConsistencyManagerDemo_tvOSUITests.swift
+//  ConsistencyManagerDemo-tvOSUITests
+//
+//  Created by Andrew Sowers on 1/6/17.
+//  Copyright © 2017 LinkedIn. All rights reserved.
+//
+
+import XCTest
+
+class ConsistencyManagerDemo_tvOSUITests: XCTestCase {
```
Comment (plivesey) [inline SampleApp/ConsistencyManagerDemo-tvOSUITests/ConsistencyManagerDemo_tvOSUITests.swift]: Can we remove these UI tests? I highly doubt we'll ever write UI tests for the demo app and it's just cruft.

**Review by plivesey — COMMENTED** [review]
Can you also make it so that we run tests on tvOS. you need to edit the `build.sh` and add another clean derived data and run tests on the right sdk. Something like:
```
rm -rf $DERIVED_DATA &&
time xcodebuild clean test \
    -project ConsistencyManager.xcodeproj \
    -scheme ConsistencyManager \
    -sdk macosx \ <--- this should be tvOS though? or something?
    -derivedDataPath $DERIVED_DATA \
    | tee build.log \
    | xcpretty &&
```

We don't test the demo app with travis btw, so just the main project.

**Comment by plivesey** [comment]
Left a couple of comments on the code too, but all minor stuff. Then can merge this in. :D

**Comment by asowers1** [comment]
I'll patch these up and push 👍

File: SampleApp/ConsistencyManagerDemo-tvOSUITests/ConsistencyManagerDemo_tvOSUITests.swift (hunk 2 above)
Comment (asowers1) [reply SampleApp/ConsistencyManagerDemo-tvOSUITests/ConsistencyManagerDemo_tvOSUITests.swift]: This should probably do it for xcodebuild: ``` rm -rf $DERIVED_DATA &&
    time xcodebuild clean build \
    -project ConsistencyManager.xcodeproj \
    -scheme ConsistencyManager \
    -sdk appletvsimulator10.0 \
    -derivedDataPath $DERIVED_DATA \
    -destination 'platform=tvOS Simulator,name=Apple TV 1080p,OS=10.0' \
    | tee build.log \
    | xcpretty && ```

**Comment by asowers1** [comment]
I'm not seeing the empty scheme for ConsistencyManager in the SampleApp project. Is it still in there after this push?

**Comment by plivesey** [comment]
I'll check on the empty scheme, but don't worry about that either way. It may be a local thing, and we can delete that later. Checking changes.

File: SampleApp/ConsistencyManagerDemo-tvOSUITests/ConsistencyManagerDemo_tvOSUITests.swift (hunk 2 above)
Comment (asowers1) [reply SampleApp/ConsistencyManagerDemo-tvOSUITests/ConsistencyManagerDemo_tvOSUITests.swift]: https://github.com/linkedin/LayoutKit/blob/master/build.sh

**Comment by plivesey** [comment]
Yep. LGTM. Merging.

**Comment by asowers1** [comment]
🎉 Thanks!

**Pull request merged**